src/main/java/org/wso2/carbon/securevault/hashicorp/repository/HashiCorpSecretRepository.java

```diff
@@ -179,4 +202,130 @@ public SecretRepository getParent() {
 
         return this.parentRepository;
     }
+
+    /**
+     * Get the root token of the vault. Either by prompting the user via the console or by accessing the text file
+     * containing the root token.
+     */
+    private void retrieveRootToken() throws HashiCorpVaultException {
+
+        String carbonHome = System.getProperty(CARBON_HOME);
+        setTextFileName();
+
+        if (new File(carbonHome + File.separator + textFilePersist).exists()) {
+            persistToken = true;
+        }
+
+        tokenFile = new File(carbonHome + File.separator + textFileName);
+        if (tokenFile.exists()) {
+            rootToken = readToken(tokenFile);
+
+            if (!persistToken) {
+                if (!renameConfigFile(textFileName_tmp)) {
+                    throw new HashiCorpVaultException("Error in renaming password config file.");
+                }
+            }
+        } else {
+            tokenFile = new File(carbonHome + File.separator + textFileName_tmp);
+            if (tokenFile.exists()) {
+                rootToken = readToken(tokenFile);
+
+                if (!persistToken) {
+                    if (deleteConfigFile()) {
+                        throw new HashiCorpVaultException("Error in deleting password config file.");
+                    }
+                }
+            } else {
+                tokenFile = new File(carbonHome + File.separator + textFilePersist);
+                if (tokenFile.exists()) {
+                    rootToken = readToken(tokenFile);
+
+                    if (!persistToken) {
+                        if (deleteConfigFile()) {
+                            throw new HashiCorpVaultException("Error in deleting password config file.");
+                        }
+                    }
+                } else {
+                    Console console;
+                    char[] token;
+                    if ((console = System.console()) != null && (token = console.readPassword("[%s]",
+                            "Enter the Root Token: ")) != null) {
+                        rootToken = String.valueOf(token);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Set the name for the text file which contains the root token.
+     * For Linux: The file name should be hashicorpRootToken-tmp/ hashicorpRootToken-persist.
+     * For Windows: The file name should be hashicorpRootToken-tmp.txt/ hashicorpRootToken-persist.txt.
```

Comment:
Shall we change "/" to "or"?" to make it more clear?

Reply:
Changed

Reply:
Changed